File: python/source_code/Python_Core_Programming_2/ch23/swhello.py
#!/usr/bin/env jython

from pawt import swing
import sys
from java.awt import Color, BorderLayout

def quit(e):
    sys.exit()

top = swing.JFrame("PySwing")
box = swing.JPanel(BorderLayout())
hello = swing.JLabel("Hello World!")
quit = swing.JButton("QUIT", actionPerformed=quit,
    background=Color.red, foreground=Color.white)

box.add("North", hello)
box.add("South", quit)
top.contentPane.add(box)
top.pack()
top.visible = 1	# or True for Jython 2.2+

File: python/source_code/Python_Core_Programming_2/ch23/swhello.java
import java.awt.*;
import java.awt.event.*;
import javax.swing.*;
import java.lang.*;

public class swhello extends JFrame {
    JPanel box;
    JLabel hello;
    JButton quit;

    public swhello() {
        super("JSwing");
        JPanel box = new JPanel(new BorderLayout());
        JLabel hello = new JLabel("Hello World!");
        JButton quit = new JButton("QUIT");

        ActionListener quitAction = new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                System.exit(0);
            }
        };
        quit.setBackground(Color.red);
        quit.setForeground(Color.white);
        quit.addActionListener(quitAction);
        box.add(hello, BorderLayout.NORTH);
        box.add(quit, BorderLayout.SOUTH);

        addWindowListener(new WindowAdapter() {
            public void windowClosing(WindowEvent e) {
                System.exit(0);
            }
        });
        getContentPane().add(box);
        pack();
        setVisible(true);
    }

    public static void main(String args[]) {
        swhello app = new swhello();
    }
}
